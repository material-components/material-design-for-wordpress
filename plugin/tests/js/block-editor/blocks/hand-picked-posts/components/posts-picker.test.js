/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import { render, fireEvent } from '@testing-library/react';

/**
 * Internal dependencies
 */
import PostsPicker from '../../../../../../assets/src/block-editor/blocks/hand-picked-posts/components/posts-picker';

// Mock PostsControl component as not relevant in this test and failing to pass tests
// due to the SearchListControl WooCommerce component.
jest.mock(
	'../../../../../../assets/src/block-editor/components/posts-control/index.js',
	() => {
		const PostsControl = () => <div />;
		return PostsControl;
	}
);

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const fullRender = props => {
	return render( <PostsPicker { ...props } /> );
};

/**
 * Shallow render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const shallowRender = props => {
	return shallow( <PostsPicker { ...props } /> );
};

describe( 'PostsPicker', () => {
	it( 'matches snapshot using full rendering', () => {
		const wrapper = fullRender( {
			attributes: { posts: [ 1, 2, 3 ] },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot using shallow rendering', () => {
		const wrapper = shallowRender( {
			attributes: { posts: [ 1, 2, 3 ] },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'invokes setAttributes with editMode on `Done` click', () => {
		const props = {
			attributes: { posts: [ 1, 2, 3 ] },
			setAttributes: jest.fn(),
			debouncedSpeak: jest.fn(),
		};
		const { getByText } = fullRender( props );

		fireEvent.click( getByText( 'Done' ) );

		expect( props.setAttributes.mock.calls[ 0 ][ 0 ].editMode ).toStrictEqual(
			false
		);
		expect( props.debouncedSpeak.mock.calls[ 0 ][ 0 ] ).toContain(
			'Showing Curated Post Collection block preview.'
		);
	} );
} );
