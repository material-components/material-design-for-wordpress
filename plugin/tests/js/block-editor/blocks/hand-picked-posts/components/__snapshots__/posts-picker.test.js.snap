// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsPicker matches snapshot using full rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="components-placeholder material-design-block-products-grid material-design-block-handpicked-posts is-small"
      >
        <iframe
          aria-hidden="true"
          frameborder="0"
          src="about:blank"
          style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
          tabindex="-1"
        />
        <div
          class="components-placeholder__label"
        >
          <i
            class="material-icons-outlined"
            size="24"
          >
            library_books
          </i>
          Curated Post Collection
        </div>
        <div
          class="components-placeholder__fieldset"
        >
          <div
            class="material-design-block-handpicked-posts__types"
          >
            <p>
              Display a selection of hand-picked posts.
            </p>
            <p
              style="font-weight: bold; text-align: left;"
            >
              Content
            </p>
          </div>
          <div
            class="material-design-block-handpicked-posts__selection"
          >
            <div />
            <button
              class="components-button is-primary"
              type="button"
            >
              Done
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="components-placeholder material-design-block-products-grid material-design-block-handpicked-posts is-small"
    >
      <iframe
        aria-hidden="true"
        frameborder="0"
        src="about:blank"
        style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
        tabindex="-1"
      />
      <div
        class="components-placeholder__label"
      >
        <i
          class="material-icons-outlined"
          size="24"
        >
          library_books
        </i>
        Curated Post Collection
      </div>
      <div
        class="components-placeholder__fieldset"
      >
        <div
          class="material-design-block-handpicked-posts__types"
        >
          <p>
            Display a selection of hand-picked posts.
          </p>
          <p
            style="font-weight: bold; text-align: left;"
          >
            Content
          </p>
        </div>
        <div
          class="material-design-block-handpicked-posts__selection"
        >
          <div />
          <button
            class="components-button is-primary"
            type="button"
          >
            Done
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PostsPicker matches snapshot using shallow rendering 1`] = `
<Placeholder
  className="material-design-block-products-grid material-design-block-handpicked-posts"
  icon={
    <i
      className="material-icons-outlined"
    >
      library_books
    </i>
  }
  label="Curated Post Collection"
>
  <div
    className="material-design-block-handpicked-posts__types"
  >
    <p>
      Display a selection of hand-picked posts.
    </p>
    <p
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "left",
        }
      }
    >
      Content
    </p>
    <RadioControl
      className="material-design-block-handpicked-posts__types__list"
      onChange={[Function]}
    />
  </div>
  <div
    className="material-design-block-handpicked-posts__selection"
  >
    <PostsControl
      onChange={[Function]}
      selected={
        Array [
          1,
          2,
          3,
        ]
      }
    />
    <ForwardRef(Button)
      isPrimary={true}
      onClick={[Function]}
    >
      Done
    </ForwardRef(Button)>
  </div>
</Placeholder>
`;
